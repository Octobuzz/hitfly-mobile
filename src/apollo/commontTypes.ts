export type CollectionsType = 'recommended' | 'musicFan'
