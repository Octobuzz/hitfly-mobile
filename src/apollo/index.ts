import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
