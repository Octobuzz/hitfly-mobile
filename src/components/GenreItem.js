import React from 'react'
import PropTypes from 'prop-types'
import {
  StyleSheet,
  View,
  Image,
  Text,
  TouchableOpacity,
  ViewPropTypes,
} from 'react-native'
import { colors, style } from '../constants'
import { hasValue } from '../utils/helpers'

/**
 * Квадратная иконка с закругленными краями и две надписи
 * @param {object} props
 * @param {string} props.title Заголовок
 * @param {string} props.description Описание
 * @param {*} props.image Ссылка на картинку вложенную или {uri: 'link'}
 * @param {number} props.size Размер блока (ширина)
 * @param {object} props.style Стиль для врапера
 * @param {func} props.onPress Функция на нажатие - при отсутсвии компонент перестает быть кликабельным
 */
const GenreItem = ({ title, description, image, size, style, onPress }) => {
  const wrapperStyle = [
    styles.wrapper,
    {
      width: size,
    },
    style,
  ]
  const imageWrapperStyle = [
    styles.imageWrapper,
    {
      width: size,
    },
  ]
  const imageStyle = [
    styles.imageWrapper,
    {
      width: size,
      height: size,
      borderRadius: 4,
    },
  ]
  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={!onPress}
      activeOpacity={0.8}
      style={wrapperStyle}
    >
      {hasValue(image) && (
        <View style={imageWrapperStyle}>
          <Image style={imageStyle} source={image} resizeMode="cover" />
        </View>
      )}
      {(hasValue(title) || hasValue(description)) && (
        <View style={styles.bottomWrapper}>
          {hasValue(title) && (
            <Text numberOfLines={2} style={styles.title}>
              {title}
            </Text>
          )}
          {hasValue(description) && (
            <Text numberOfLines={1} style={styles.description}>
              {description}
            </Text>
          )}
        </View>
      )}
    </TouchableOpacity>
  )
}

GenreItem.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  onPress: PropTypes.func,
}

GenreItem.defaultProps = {
  size: 100,
}

const styles = StyleSheet.create({
  wrapper: {},
  imageWrapper: {
    overflow: 'hidden',
    alignSelf: 'center',
    justifyContent: 'flex-end',
  },
  image: {},
  bottomWrapper: {
    paddingTop: 6,
  },
  title: {
    ...style.text.medium,
    color: colors.BLACK_LABEL,
    fontSize: 14,
    lineHeight: 16,
    paddingTop: 2,
  },
  description: {
    ...style.text.regular,
    color: colors.BLACK_LABEL,
    fontSize: 12,
    lineHeight: 14,
    paddingTop: 4,
  },
})

export default GenreItem
