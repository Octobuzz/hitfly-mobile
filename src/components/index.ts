export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
