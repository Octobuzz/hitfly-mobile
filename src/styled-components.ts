import * as styledComponents from 'styled-components/native'

export interface ThemeColors {
  brandBlue: string
  brandPink: string
  white: string
  textMain: string
  textAlt: string
  textGray: string
  inputBorder: string
}

export interface ThemeFonts {
  black: string
  bold: string
  medium: string
  regular: string
}

export interface ITheme {
  colors: ThemeColors
  fonts: ThemeFonts
}

const {
  default: styled,
  css,
  ThemeProvider,
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<
  ITheme
>

export { css, ThemeProvider }
export default styled
