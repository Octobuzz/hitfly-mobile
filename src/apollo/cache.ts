import { InMemoryCache } from 'apollo-cache-inmemory'
import { persistCache } from 'apollo-cache-persist'
import { storageInstance } from 'src/utils'
import gql from 'graphql-tag'

export const defaults = {
  headerSettings: {
    mode: 'dark',
    state: 'main',
    __typename: 'HeaderSettings',
  },
  isPlaying: false,
  activeTrack: null,
  playlist: [],
}

export const typeDefs = gql`
  type Track {
    id: Int!
    title: String
    singer: String
    date: String
    songText: String
    fileUrl: String
    musicWave: [Int]
    length: Int
  }
  type HeaderSettings {
    style: String!
    state: String!
  }
`

export default async (): Promise<InMemoryCache> => {
  const cache = new InMemoryCache({
    freezeResults: true,
  })

  await persistCache({
    cache,
    debug: __DEV__,
    // @ts-ignore
    storage: storageInstance,
  })

  return cache
}
