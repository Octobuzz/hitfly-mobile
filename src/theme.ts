import { ITheme } from './styled-components'

const theme: ITheme = {
  colors: {
    brandBlue: '#4297FE',
    brandPink: '#D066BF',
  },
}

export default theme
