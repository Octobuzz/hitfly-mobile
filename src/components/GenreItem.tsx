import React from 'react'
import FastImage from 'react-native-fast-image'
import { images } from 'src/constants'
import styled from 'src/styled-components'
import TextBase from 'src/components/TextBase'
import { Genre } from 'src/apollo'

const ITEM_SIZE = 109

const Wrapper = styled.TouchableOpacity`
  border-radius: 4px;
  width: ${ITEM_SIZE}px;
  height: ${ITEM_SIZE}px;
  overflow: hidden;
  align-items: center;
  justify-content: center;
`

const CornerImage = styled.Image.attrs(({ isSelected }: Selectable) => ({
  source: isSelected ? images.GENRE_ACTIVE : images.GENRE_INACTIVE,
}))<Selectable>`
  position: absolute;
  top: 0;
  right: 0;
`

const GenreImage = styled(FastImage)`
  width: 100%;
  height: 100%;
`

const TitleText = styled(TextBase)`
  position: absolute;
  text-align: center;
  font-family: ${({ theme }) => theme.fonts.bold};
  color: ${({ theme }) => theme.colors.white};
`

interface Selectable {
  isSelected?: boolean
  isSelectable?: boolean
}

interface Props extends Selectable {
  item: Genre
  onPress: (item: Genre) => void
}

interface Sized {
  size: number
}

const GenreItem: React.FC<Props> & Sized = ({
  item,
  onPress,
  isSelected,
  isSelectable,
}) => {
  const { imageUrl, title } = item
  const handlePress = React.useCallback(() => {
    onPress(item)
  }, [onPress, item])

  return (
    <Wrapper onPress={handlePress}>
      <GenreImage source={{ uri: imageUrl }} />
      {isSelectable && <CornerImage isSelected={isSelected} />}
      <TitleText>{title}</TitleText>
    </Wrapper>
  )
}

GenreItem.size = ITEM_SIZE

export default GenreItem
