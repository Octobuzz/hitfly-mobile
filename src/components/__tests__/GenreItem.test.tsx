import React from 'react'
import GenreItem from '../GenreItem'
import { Genre } from 'src/apollo'
import { ThemeProvider } from 'src/styled-components'
import renderer from 'react-test-renderer'
import theme from 'src/theme'

test('renders correctly', () => {
  const item = { title: 'title', imageUrl: 'url' } as Genre
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <GenreItem item={item} />
      </ThemeProvider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
