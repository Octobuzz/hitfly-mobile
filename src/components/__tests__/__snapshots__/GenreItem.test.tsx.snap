// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "height": 109,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 109,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "GothamPro-Bold",
          "fontSize": 16,
          "lineHeight": 20,
          "position": "absolute",
          "textAlign": "center",
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
